packages/dos-donts-block/src/DosDontsBlock.tsx

```diff
@@ -1,95 +1,458 @@
+    const [minRowHeight, setMinRowHeight] = useState(60);
+    const [isUploadLoading, setIsUploadLoading] = useState(false);
+    const [selectedType, setSelectedType] = useState<DoDontType | undefined>();
+    const [selectedAssets, setSelectedAssets] = useState<Asset[] | undefined>();
+    const [openFileDialog, { selectedFiles: filesFromInput }] = useFileInput({ multiple: true, accept: 'image/*' });
+    const [selectedFiles, setSelectedFiles] = useState<FileList | null>(filesFromInput);
+    const [uploadFile, { results: uploadResults, doneAll }] = useAssetUpload({
+        onUploadProgress: () => !isUploadLoading && setIsUploadLoading(true),
+    });
+    const [activeId, setActiveId] = useState<string | undefined>(undefined);
     const {
+        style,
+        hasCustomDoIcon,
+        doIconChoice,
+        hasCustomDontIcon,
+        dontIconChoice,
+        hasStrikethrough,
+        columns,
+        keepSideBySide,
+        isCustomColumnGutter,
+        customColumnGutterValue,
+        columnGutterChoice,
+        isCustomRowGutter,
+        customRowGutterValue,
+        rowGutterChoice,
+        isCustomImageHeight,
+        customImageHeightValue,
+        imageDisplay,
+        imageHeightChoice,
+        doColor: customDoColor,
+        dontColor: customDontColor,
+        hasCustomDoColor,
+        hasCustomDontColor,
         items = [],
-        columns = 2,
-        isCustomSpacing = false,
-        spacingValue = '',
-        doColor = DO_COLOR_DEFAULT_VALUE,
-        dontColor = DONT_COLOR_DEFAULT_VALUE,
-        layout = DoDontLayout.SideBySide,
-        style = DoDontStyle.Icons,
-        spacingChoice = DoDontSpacing.Medium,
+        mode,
+        borderWidth,
+        backgroundColor,
+        borderColor,
+        borderStyle,
+        hasBackground,
+        hasBorder,
+        hasRadius,
+        radiusChoice,
+        radiusValue,
     } = blockSettings;
+    const sensors = useSensors(useSensor(PointerSensor));
+    const { dontIconAsset, doIconAsset, itemImages } = blockAssets;
     const [localItems, setLocalItems] = useState<Item[]>(items);
+    const doColor = hasCustomDoColor ? customDoColor : DO_COLOR_DEFAULT_VALUE;
+    const dontColor = hasCustomDontColor ? customDontColor : DONT_COLOR_DEFAULT_VALUE;
 
-    const setItems = (numberOfItems: number) => {
-        let updatedItems = localItems;
-        // Check whether to add or remove items
-        if (updatedItems.length - numberOfItems < 0) {
-            for (let index = updatedItems.length; index < numberOfItems; index++) {
-                updatedItems.push({ id: index, title: '', body: '' });
-            }
-        } else {
-            updatedItems = updatedItems.slice(0, numberOfItems);
+    /**
+     * Create placeholders on mount if empty
+     */
+    useEffect(() => {
+        if (localItems.length === 0) {
+            const placeholderItems: Item[] = [
+                {
+                    id: cuid(),
```

Comment:
You can remove the lib and use `window.crypto.randomUUID()`

Reply:
I think this would not work in the following scenario.
[1,2,3,4] are the current items. [3] is deleted, and then I add a new item, the new ID list would be [1,2,4,4].

However I removed CUID and reused the `generateRandomId` from fondue.